### PR TITLE
Enable code analysis and address initial set of issues.

### PR DESCRIPTION
**Pull request by adityamandaleeka**
Supersedes #58167.

This continues work @BrennanConroy started, with some modifications. Most notably, C26477 (nullptr) has been handled. I also made an effort to isolate different kinds of changes in each commit to make reviewing (and potential future reverts) easier.

We'll continue to work on this area, but it doesn't make sense to delay reviewing this any longer so here it is.

### CODE REVIEW

File: src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/dllmain.cpp [hunk 1]
```diff
@@ -151,12 +151,16 @@ EnsureOutOfProcessInitializtion(IHttpApplication *pHttpApplication)
             }
         }
 
+#pragma warning(push)
+#pragma warning(disable: 26477) // NULL usage via Windows header
```
[the remark this answers is not in the document]
Comment (adityamandaleeka) [reply src/Servers/IIS/AspNetCoreModuleV2/OutOfProcessRequestHandler/dllmain.cpp:155]: `WINHTTP_NO_PROXY_NAME` and `WINHTTP_NO_PROXY_BYPASS` are defined as NULL in winhttp.h.

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLib/debugutil.cpp [hunk 2]
```diff
@@ -114,7 +114,7 @@ std::wstring
 GetModuleName()
 {
     WCHAR path[MAX_PATH];
-    LOG_LAST_ERROR_IF(!GetModuleFileName(g_hModule, path, sizeof(path)));
+    LOG_LAST_ERROR_IF(!GetModuleFileName(g_hModule, path, MAX_PATH));
```
[the remark this answers is not in the document]
Comment (adityamandaleeka) [reply src/Servers/IIS/AspNetCoreModuleV2/CommonLib/debugutil.cpp:117]: Sure, updating to `_countof`.

File: src/Servers/IIS/AspNetCoreModuleV2/IISLib/stringa.cpp [hunk 3]
```diff
@@ -464,6 +464,7 @@ Return Value:
 
     HRESULT hr = SafeVsnprintf(pszFormatString, argsList);
 
+#pragma warning(suppress: 26477)
```
[the remark this answers is not in the document]
Comment (adityamandaleeka) [reply src/Servers/IIS/AspNetCoreModuleV2/IISLib/stringa.cpp:473]: Done. Though an upcoming PR is going to delete this code ;)

File: src/Servers/IIS/AspNetCoreModuleV2/DefaultRules.ruleset [hunk 4]
```diff
@@ -80,7 +80,6 @@
     <Rule Id="C26475" Action="Error" />
     <Rule Id="C26476" Action="None" /> <!-- Expression/symbol 'name' uses a naked union 'union' with multiple type pointers: Use variant instead -->
     <Rule Id="C26477" Action="Error" />
-    <Rule Id="C26478" Action="None" /> <!-- Don't use std::move on constant variables. -->
```
[the remark this answers is not in the document]
Comment (adityamandaleeka) [reply src/Servers/IIS/AspNetCoreModuleV2/DefaultRules.ruleset]: Good catch. I addressed the issues but deleted this instead of leaving it as `Error`.

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**